x/distribution/types/dec_coin.go

```diff
@@ -0,0 +1,141 @@
+package types
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Coins which can have additional decimal points
+type DecCoin struct {
+	Denom  string  `json:"denom"`
+	Amount sdk.Dec `json:"amount"`
+}
+
+func NewDecCoin(coin sdk.Coin) DecCoin {
```

Comment:
`NewDecCoinFromCoin` perhaps - do we want another constructor which takes `sdk.Dec` ?

Reply:
yup to both